### PR TITLE
Improved `_got_announcement`

### PR DESCRIPTION
**Pull request by exarkun**
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3460

### CODE REVIEW

File: src/allmydata/storage_client.py [hunk 1]
```diff
@@ -292,28 +306,71 @@ def _check_connected_high_water_mark(self):
...
+        :param bytes server_id: The unique identifier for the storage server
+            which made the announcement.
+
+        :param dict ann: The announcement.
+
+        :return bool: ``True`` if the announcement should be ignored,
+            ``False`` if it should be used to update our local storage server
+            state.
+        """
+        # Let local static configuration always override any announcement for
```
Comment (meejah) [inline src/allmydata/storage_client.py:323]: This new method seems to drop two relevant checks: "starts with `v0-`" and "is announcing storage-service".

`precondition` looks an awful lot like `assert` but will actually run all the time as far as I can tell and these checks seem worth saving (unless there's good reasons to drop them?)

**Review by meejah — APPROVED** [review]
Overall LGTM :+1: but see one inline comment.

These days I do always ask myself "should it not say ignore?" .. that is, does it make sense without a double-negative? That is, maybe `_should_accept_announcement()` and a non-early-return version of the main method? Not sure if this really helps readability that much here though.

**Comment by exarkun** [comment]
> These days I do always ask myself "should it not say ignore?" .. that is, does it make sense without a double-negative? 

Yea I try to keep an eye out for this as well.  I feel okay about how it is now because "ignore" is kind of an affirmative thing, not just "not accept", and the calling code doesn't use `not` with the result.

**Comment by exarkun** [comment]
Fixes: ticket:3460

**Pull request merged**